lib/benchee/benchmark.ex

```diff
@@ -26,14 +26,23 @@ defmodule Benchee.Benchmark do
 
   Warmup is usually important for run times with JIT but it seems to have some
   effect on the BEAM as well.
+
+  A parallel amount of processes will be spawned, each executing bench functions.
   """
-  def measure(suite = %{jobs: jobs, config: %{time: time, warmup: warmup}}) do
+  def measure(suite = %{jobs: jobs, config: %{parallel: parallel, time: time, warmup: warmup}}) do
     run_times = Enum.map jobs, fn({name, function}) ->
-                  IO.puts "Benchmarking #{name}..."
-                  run_warmup function, warmup
-                  job_run_times = measure_runtimes function, time
-                  {name, job_run_times}
-                end
+      IO.puts "Benchmarking name: #{name}, parallel: #{parallel}, time: #{time / 1_000_000}, warmup: #{warmup / 1_000_000}..."
+      job_run_times =
+        1..parallel
+        |> Enum.map(fn(_) ->
+             Task.async(fn ->
+               run_warmup function, warmup
+               measure_runtimes function, time
+             end)
+           end)
```

Comment:
for readabilities sake I'd probably take this code and put it into a separate function :)